pkg/observability/stackdriver.go

```diff
@@ -129,23 +128,30 @@ func (e *stackdriverExporter) StartExporter(ctx context.Context) error {
 		descriptorCreateRequests = append(descriptorCreateRequests, cmrdesc)
 	}
 
-	// register metrics in the background and don't block server startup on failures.
+	// Register metrics in the background and don't block server startup on
+	// failures.
+	//
+	// Future optimization: only register metrics once per buildinfo data.
 	go func() {
-		// Create a separate timeout for metric registration.
-		ctx, done := context.WithTimeout(context.Background(), time.Duration(5*len(descriptorCreateRequests))*time.Second)
+		// Overall metrics registration can take a maximum of 30s.
```

Comment:
I had this higher - because the 5s timeout on individual metrics is larger than 30s in practice it probably doesn't matter

Reply:
It was 5s * number of metrics, which is identical to each having a max of 5 seconds, so I think it was strictly redundant?